### PR TITLE
Search test

### PR DESCRIPTION
**Pull request by clare**
Added a test that tests get by tag, and get by tag type.

### CODE REVIEW

File: test/testrecording.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        self._testrecording = testrecording
+        self._tagData = {"animal": animal}
+        print("  which is tagged as a {}".format(animal), end='')
+
+    def by(self, testuser):
+        print(" by {}".format(testuser.username))
+        testuser.tag_recording(self._testrecording.recordingId, self._tagData)
+        return self._testrecording
+
+    def byAI(self, testuser):
```
Comment (mjs) [inline test/testrecording.py]: It does this take `testuser`? It doesn't use it.

Maybe call this method `is_automatic()` or something?

File: test/testuser.py [hunk 2]
```diff
@@ -6,18 +6,59 @@ def __init__(self, username, userapi):
...
-        recordings = self._userapi.query(limit=1)
+    def when_searching_with_tagmode(self, tagmode):
+        queryParams = {"tagmode": tagmode}
+        return RecordingQueryPromise(self, queryParams)
+
+    def when_searching_for_tags(self, *tags):
+        queryParams = {"tags": tags}
+        return RecordingQueryPromise(self, queryParams)
+
+    def can_see_recordings(self, *expectedTestRecordings):
```
Comment (mjs) [inline test/testuser.py]: The convention in python is to use lower_case_with_underscores for variables names, not camelCase.

File: test/userapi.py [hunk 3]
```diff
@@ -84,6 +84,11 @@ def get_user_details(self, username):
         response.raise_for_status()
         # print(response.json())
         
+    def tag_recording(self, recordingId, tagDictionary):
+        url = urljoin(self._baseurl, "/api/v1/tags/")
+        tagData = {"tag": json.dumps(tagDictionary), "recordingId": recordingId}
+        response = requests.post(url, headers=self._auth_header, data=tagData)
+        response.raise_for_status()
```
Comment (mjs) [inline test/userapi.py]: use check_response instead?

File: test/testrecording.py (hunk 1 above)
Comment (clare) [reply test/testrecording.py]: The code for doing this is all in userapi, so needs a user.   Without a user you have to somehow get the URL etc so I just took a short cut and get the caller to pass in the admin user.   I now it's not strictly necessary.   Feel free to change it.

**Pull request merged**